notifier/src/main/java/com/expedia/adaptivealerting/notifier/service/Notifier.java

```diff
@@ -0,0 +1,107 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ObjectUtils;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+@Component
+@Slf4j
+public class Notifier implements ApplicationListener<ApplicationReadyEvent> {
+
+    private long TIME_OUT = 10_000;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private final KafkaConsumer<String, MappedMetricData> kafkaConsumer;
+    private final RestTemplate restTemplate;
+    private final String webhookUrl;
+
+    @Autowired
+    public Notifier(AppConfig appConfig, RestTemplate restTemplate) {
+        kafkaConsumer = buildKafkaConsumer(appConfig);
+        this.restTemplate = restTemplate;
+        this.webhookUrl = appConfig.getWebhookUrl();
+    }
+
+    private KafkaConsumer<String, MappedMetricData> buildKafkaConsumer(AppConfig appConfig) {
+        KafkaConsumer<String, MappedMetricData> kafkaConsumer = new KafkaConsumer<>(appConfig.getKafkaConsumerConfig());
+        kafkaConsumer.subscribe(Arrays.asList(appConfig.getKafkaTopic()));
+        return kafkaConsumer;
+    }
+
+    @Override
+    public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
+        processAlerts();
+    }
+
+    private void processAlerts() {
+        while(!ObjectUtils.isEmpty(webhookUrl)) {
+            ConsumerRecords<String, MappedMetricData> consumerRecords = kafkaConsumer.poll(TIME_OUT);
+            consumerRecords.forEach( record -> {
+                buildJson(record.value()).ifPresent(json -> {
+                    HttpHeaders headers = new HttpHeaders();
+                    headers.setContentType(MediaType.APPLICATION_JSON);
+                    HttpEntity<String> entity = new HttpEntity<>(json, headers);
+                    try {
+                        ResponseEntity responseEntity =
+                            restTemplate.exchange(webhookUrl, HttpMethod.POST, entity, ResponseEntity.class);
+                        if (responseEntity.getStatusCode().is2xxSuccessful()) {
+                            MetricsMonitor.notification_success.mark();
```

Comment:
meter to track failures and success. i think probably we should add counters/meters in other AA apps to track the data flow, success and failures.

Reply:
Good call